src/utils/flattenMetadataQueryResponse.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @flow
+ * @file File for some simple dom utilities
+ * @author Box
+ */
+
+import omit from 'lodash/omit';
+import get from 'lodash/get';
+import type {
+    MetadataQueryResponse,
+    MetadataQeuryResponseEntry,
+    MetadataQeuryResponseEntryMetadata,
+} from '../common/types/metadataQueries';
+import { ITEM_TYPE_FILE } from '../common/constants';
+
+type flattenedMetadataResponseEntry = {
+    id?: string,
+    metadata?: {
+        data: Object,
+        id: string,
+        metadataTemplate: {
+            templateKey: string,
+            type: string,
+        },
+    },
+    name?: string,
+    size?: number,
+};
+
+const convertMetadata = (metadata: MetadataQeuryResponseEntryMetadata) => {
+    let md = {};
+
+    Object.keys(metadata).forEach(scope => {
+        Object.keys(metadata[scope]).forEach(templateKey => {
```

Comment:
Is there a limit to the amount of metadata scopes returned as to not have an unforeseen performance penalty here?

Reply:
This is a concern @jstoffan had raised earlier. Typical metadata_queries response is of structure:
`
{
    entries: [
        item: {...},
        metadata: {
            enterprise_<eid>: {
                <templateKey>: {...}
            }
        }
    ],
    next_marker: "markermarkermarker"
}
`

Because both `enterprise_<eid>` and `<templateKey>` are dynamic key values, getting the actual metadata through `Object.keys` is one way of doing it.